### PR TITLE
Vks4git/server prototype

### PR DESCRIPTION
**Pull request by vks4git**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (vlasin) [inline LICENSE]: This one will use BSL 1.1 license: the code can be used for commercial purposes only after a certain date.

See https://github.com/zkFold/p2p-onramp?tab=License-1-ov-file#readme

File: smart-contract-wallet.cabal [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+name:               smart-contract-wallet 
+version:            0.1.0.0
+tested-with:        GHC == 9.6.3
+synopsis:           ZkFold Smart contract wallet server 
+homepage:           https://zkfold.io/
+category:           Cryptography, DSL, Math, Symbolic Computation
+author:             Vladimir Sinyakov, Pavel Sokolov, Vladimir Morozov, Eitan Chatav, Ilya Baryshnikov, Anna Hovanskaya, Evgenii Samodelov, Murat Kasimov, Maxim Churilov
+maintainer:         vlasin@zkfold.io
+license:            MIT
```
Comment (vlasin) [inline smart-contract-wallet.cabal]: BSL 1.1

File: smart-contract-wallet.cabal [hunk 3]
```diff
@@ -0,0 +1,135 @@
+cabal-version:      3.0
+
+name:               smart-contract-wallet 
+version:            0.1.0.0
+tested-with:        GHC == 9.6.3
+synopsis:           ZkFold Smart contract wallet server 
+homepage:           https://zkfold.io/
+category:           Cryptography, DSL, Math, Symbolic Computation
+author:             Vladimir Sinyakov, Pavel Sokolov, Vladimir Morozov, Eitan Chatav, Ilya Baryshnikov, Anna Hovanskaya, Evgenii Samodelov, Murat Kasimov, Maxim Churilov
```
Comment (vlasin) [inline smart-contract-wallet.cabal]: You can include just yourself for now. We will edit the list based on the contributions.

File: smart-contract-wallet.cabal [hunk 4]
```diff
@@ -0,0 +1,135 @@
+cabal-version:      3.0
+
+name:               smart-contract-wallet 
+version:            0.1.0.0
+tested-with:        GHC == 9.6.3
+synopsis:           ZkFold Smart contract wallet server 
+homepage:           https://zkfold.io/
+category:           Cryptography, DSL, Math, Symbolic Computation
```
Comment (vlasin) [inline smart-contract-wallet.cabal]: Please select more suitable categories here.

File: src/ZkFold/Wallet/API.hs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+module ZkFold.Wallet.API where
```
Comment (vlasin) [inline src/ZkFold/Wallet/API.hs]: I would name this module `ZkFold.Wallet.SymbolicAPI`

File: src/ZkFold/Wallet/Server/Types.hs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+module ZkFold.Wallet.Server.Types where
+
+import           Data.Text           (Text)
+import           Servant
+
+import           ZkFold.Wallet.Types
+
+type WalletAPI =
```
Comment (vlasin) [inline src/ZkFold/Wallet/Server/Types.hs:8]: This is actually CIP30 Wallet API. We will need to implement these functions to complete M2. "Symbolic Wallet API" gives some additional methods that we can complete later.

Comment (vks4git) [reply src/ZkFold/Wallet/Server/Types.hs:8]: Oh, so basic logic from [this issue](https://github.com/zkFold/smart-contract-wallet/issues/3) is actually all these functions? Gotcha

**Review by vlasin — APPROVED** [review]

**Pull request merged**